src/main/java/com/giovds/AverageAgeMojo.java

```diff
@@ -0,0 +1,133 @@
+package com.giovds;
+
+import com.giovds.dto.DependencyResponse;
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Mojo(name = "average",
+        defaultPhase = LifecyclePhase.VERIFY,
+        requiresOnline = true,
+        requiresProject = true,
+        requiresDependencyResolution = ResolutionScope.COMPILE)
+public class AverageAgeMojo extends AbstractMojo {
```

Comment:
When you run the `package` goal you'll get a warning that the class (and if there are any other methods and classes) undocumented with JDocs. This is required for releasing to Maven Central (https://central.sonatype.org/publish/requirements/#why-do-we-have-requirements). I didn't mention it in the CONTRIBUTING.md so that was not as transparant.

Reply:
Javadoc added - here and to the no-args constructor a few lines below.